### PR TITLE
[FIX] payment: prevent changing the company when transactions exist

### PR DESCRIPTION
**Pull request by AntoineVDV**
opw-3696841

### CODE REVIEW

File: addons/payment/models/payment_provider.py [hunk 1]
```diff
@@ -322,6 +322,20 @@ def _onchange_state_warn_before_disabling_tokens(self):
...
+    @api.onchange('company_id')
+    def _onchange_company_block_if_existing_transactions(self):
+        """ Raise a user error when the company is changed and linked transactions exist.
+
+        :return: None
+        :raise UserError: If transactions are linked to the provider.
+        """
+        if self._origin.company_id != self.company_id and self.env['payment.transaction'].search(
+            [('provider_id', '=', self._origin.id)], limit=1
+        ):
```
Comment (Demesmaeker) [inline addons/payment/models/payment_provider.py:334]: Why `self._origin.id`? :thinking:

**Review by Demesmaeker — COMMENTED** [review]
Hello :duck: 

If you need to change something, then maybe also use "transaction" instead of "transactions" as even a single one would be enough.

File: addons/payment/models/payment_provider.py (hunk 1 above)
Comment (AntoineVDV) [reply addons/payment/models/payment_provider.py:334]: Because of the @api.onchange

**Comment by AntoineVDV** [comment]
> If you need to change something, then maybe also use "transaction" instead of "transactions" as even a single one would be enough.

"transactions" is the correct way to put it. In French you'd go for the singular, but in English you use the plural form when there is a chance that there can be multiple something in "no <something>" or "zero <something>".

**Review by Demesmaeker — APPROVED** [review]

**Comment by AntoineVDV** [comment]
@robodoo r+

**Pull request closed without merging**